x-pack/plugins/canvas/common/lib/filters.js

```diff
@@ -13,15 +13,16 @@ export function time(filter) {
   if (!filter.column) {
     throw new Error('column is required for Elasticsearch range filters');
   }
+  const { from, to, column, filterGroup: group } = filter;
   return {
-    range: {
-      [filter.column]: { gte: filter.from, lte: filter.to },
-    },
+    group,
+    range: { [column]: { gte: from, lte: to } },
```

Comment:
@crob611, I'm sorry, I don't really get your idea. Please, fix me if I'm wrong.

I'm sure, this code is identical to the one we've used before with a small change - a group was added. `from` and `to` can be not only timestamps, but also `now`, `1-day`, etc.

Reply:
Ahh yes, everything does appear identical.  

I was confused because I was comparing the debug output of a `select_filter` 

Previously it would look like this 

![image](https://user-images.githubusercontent.com/515272/148843772-0244a789-6be8-4f87-a5cd-eb8a0ad2d683.png)

and with these changes it now is like 
![image](https://user-images.githubusercontent.com/515272/148843804-43bc1b47-2201-4003-84fc-d5ecf8be6a84.png)

Any idea where or how that `timeRange` is being set?

Reply:
While creating the searchContext, we are passing `timeRange` to the context of execution. https://github.com/elastic/kibana/blob/7ce9a8edb59b4ca90630a18cc206d55701b6d118/x-pack/plugins/canvas/public/services/kibana/expressions.ts#L120


And after, at `kibana` expression function we are adding it as a `timeRange` field of the `kibana_context`. https://github.com/elastic/kibana/blob/7ce9a8edb59b4ca90630a18cc206d55701b6d118/src/plugins/data/common/search/expressions/kibana.ts#L46